### DISCUSSION TITLE
Contained content items and document size

### DISCUSSION THREAD

**Discussion by NickDarvey** [OrchardCMS/OrchardCore#8627] [Q&A] [answered]
[The docs](https://docs.orchardcore.net/en/dev/docs/reference/modules/Autoroute/#contained-definition) say that:
>Contained content items are stored as part of the json inside the container document.
and that makes sense for things like taxonomies.

However, wouldn't something like a blog grow to be a very large document if each blog post (contained content item) is stored inside the container document (container)?
https://github.com/OrchardCMS/OrchardCore/blob/a27f2de7b8889e1a7649c23b819ff372fbcb8aa1/src/OrchardCore.Themes/TheBlogTheme/Recipes/blog.recipe.json#L340-L342

Am I misunderstanding something?

**Comment by deanmarcussen** [comment, marked as answer]
The docs refer to `BagParts` and `Taxonomies` where content is contained inside the json of a single Content Item.

The `Blog` uses the `ListPart` where each `BlogPost` is a single document in the database. It contains a reference to the `ContentItemId` of the `ListPart`, i.e. `Blog` which contains it.

So no, not an issue, but I can understand the confusion.

The term `Contained` is used somewhat interchangably within Orchard Core, more to refer to the concept of a `Container`, not its implementation detail.

**Reply by NickDarvey** [reply]
Ahh, I understand. Thank you @deanmarcussen
